posts/benchmarking_design_decisions/index.qmd

```diff
@@ -0,0 +1,228 @@
+---
+title: "Using benchmarking to guide the adoption of dependencies in R packages"
+author:
+  - name: "James Mba Azam"
+    orcid: "https://orcid.org/0000-0001-5782-7330"
+date: last-modified
+categories: [benchmarking, R, tidyverse, cli, R packages, software architecture]
+format:
+  html:
+    toc: true
+execute: 
+  cache: true
+---
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(
+  cache = TRUE,
+  message = FALSE
+)
+```
+
+R package developers are constantly faced with design decisions about user
+friendliness. Such design decisions often include checking that inputs are
+rightly specified and how to fix or signal issues arising from intermediate
+calculations or processes within a function (condition handling).
+
+These design decisions often result in taking on dependencies. Dependencies are
+external packages that enable a function to work properly. For example, the
+`tidyverse` is a collection of packages that are often used together to
+make data analysis easier. The `tidyverse` is a dependency for many
+packages that are built on top of it.
+
+Dependencies should not be adopted blindly without carefully considering the
+overhead they introduce. Overhead here refers to the additional work that a
+package must do to make the dependency work. In this article, we will
+demonstrate how to use (micro)benchmarking[^1] to guide the decision to take on
```

Comment:
Looks like you opted to use the bench package in the end so this note should probably be removed.

Reply:
Not sure I understand. I'm not yet referring to any package in this paragraph.

Reply:
Oh, I misunderstood you intent then. I took this was a foreshadowing of the fact you were going to use the microbenchmark package.
I now understand was to refer to the concept. Feel free to disregard this comment then.